### PR TITLE
[compatibility] fixed tensor parallel compatibility with torch 1.9

### PR DESCRIPTION
**Pull request by FrankLeeeee**
This fixed the issue #698 . As the CI uses PyTorch 1.10, I have pasted the screenshot for the unit testing `test_layers`.

Screenshot 2022-04-08 at 2 00 49 PM

### CODE REVIEW

File: colossalai/nn/layer/parallel_2d/_operation.py [hunk 1]
```diff
@@ -76,7 +77,7 @@ def forward(
         pipeline_parallel_size: int,
         tensor_parallel_size: int,
     ) -> Tensor:
-
+        A = A.clone().detach()
```
Comment (ver217) [inline colossalai/nn/layer/parallel_2d/_operation.py:80]: Do you try `A = A.detach()`?

Comment (FrankLeeeee) [reply colossalai/nn/layer/parallel_2d/_operation.py:80]: I didn't try it. However, I think `A.detach()` will lead to incorrect computation graph because `A` is the input tensor of the linear layer. If `A` is detached from the graph, the graph will be disconnected.

**Review by ver217 — APPROVED** [review]

File: colossalai/nn/layer/parallel_2d/_operation.py (hunk 1 above)
Comment (FrankLeeeee) [reply colossalai/nn/layer/parallel_2d/_operation.py:80]: According to the PyTorch documentation, `detach` will create a tensor which shares the storage with the original tensor. If there happens to be some in-place operations in the forward and backward functions, there will be incorrect arithmetic results.

Screenshot 2022-04-11 at 1 40 14 PM

Comment (ver217) [reply colossalai/nn/layer/parallel_2d/_operation.py:80]: OK
